lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1076,14 +1076,26 @@ static void optimizeSliceOfSplat(Function *F) {
 }
 
 /// Eliminate ReshapeNode when the input is already the correct shape.
+/// Also eliminate ReshapeNode when the input is SplatNode.
 static void optimizeReshape(Function *F) {
   for (auto &node : F->getNodes()) {
     auto *reshapeNode = dyn_cast<ReshapeNode>(&node);
     if (!reshapeNode)
       continue;
-    auto inputNode = reshapeNode->getInput();
+    auto &inputNode = reshapeNode->getNthInput(0);
     if (inputNode.dims() == reshapeNode->dims()) {
       reshapeNode->getResult().replaceAllUsesOfWith(inputNode);
+      continue;
+    }
+    // Reshape(Splat(args)) -> Splat(args').
+    auto *splatNode = dyn_cast<SplatNode>(inputNode);
+    if (splatNode && splatNode->hasOneUse()) {
+      // Splat node with more than one use can not be transformed, otherwise it
+      // would cause type inconsistency.
```

Comment:
This is wrong. It has nothing to do with any type inconsistency, as we create a new splat and do not touch the old one. So, types will be consistent.

The reason is that if we do it for all reshape(splats), we may increase the number of splats, which may lead to increased memory consumption during the execution of the NN model.

Reply:
Thx for the correction. Changed to the correct comment.